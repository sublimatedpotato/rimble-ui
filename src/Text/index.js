import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { display } from 'styled-system';
import { COMMON, TYPOGRAPHY } from '../constants';
import theme from '../theme';
import Box from '../Box';

export const caps = props =>
  props.caps ? { textTransform: 'uppercase' } : null;

export const italic = props => (props.italic ? { fontStyle: 'italic' } : null);

const Text = styled(Box)`
  ${TYPOGRAPHY}
  ${COMMON}
  ${display}

  ${caps}
  ${italic}
`;

Text.defaultProps = {
  theme,
  color: 'copyColor',
  fontFamily: 'sansSerif',
  fontSize: 2,
  fontWeight: 2,
  lineHeight: 'copy',
  textAlign: 'left',
};

Text.propTypes = {
  ...COMMON.propTypes,
  ...TYPOGRAPHY.propTypes,
  ...display.propTypes,
  theme: PropTypes.object,
};

Text.displayName = 'Text';

Text.span = styled(Text)``;
Text.span.defaultProps = {
  ...Text.defaultProps,
  as: 'span',
};

Text.p = styled(Text)``;
Text.p.defaultProps = {
  ...Text.defaultProps,
  as: 'p',
};

Text.s = styled(Text)``;
Text.s.defaultProps = {
  ...Text.defaultProps,
  as: 's',
};

export default Text;
