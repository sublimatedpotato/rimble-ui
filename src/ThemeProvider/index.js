import React from 'react';
import PropTypes from 'prop-types';

import styled, {
  createGlobalStyle,
  ThemeProvider as StyledThemeProvider,
} from 'styled-components';

import { normalize, transparentize } from 'polished';

import theme from '../theme';

export const Base = styled.div`
  & {
    font-size: 1em;
    font-family: ${props => props.theme.fonts.sansSerif};
    color: ${props => props.theme.colors.copyColor};
  }
`;

const GlobalStyle = createGlobalStyle`
  ${normalize()}
  html {
    box-sizing: border-box;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  ::selection {
    background: ${props => transparentize(0.7, props.theme.colors.primary)};
  }
`;

const ThemeProvider = ({ ...props }) => {
  return (
    <StyledThemeProvider theme={theme} {...props}>
      <React.Fragment>
        <GlobalStyle />
        <Base {...props} />
      </React.Fragment>
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {};

export default ThemeProvider;
