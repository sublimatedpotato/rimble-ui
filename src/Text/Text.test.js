import React from 'react';
import renderer from 'react-test-renderer';
import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Text from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

configure({ adapter: new Adapter() });

describe('Text component sanity', () => {
  it('has name', () => {
    expect(Text.displayName).toBe('Text');
  });

  it('matches default snapshot', () => {
    const component = render(<Text />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Text />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});

// describe('Text style tests', () => {
//   it('static passes display prop', () => {
//     const tree = renderer.create(<Text.span display="block" />).toJSON();

//     expect(tree).toHaveStyleRule('display', 'block');
//   });

//   it('renders with theme font family', () => {
//     const tree = renderer.create(<Text.span />).toJSON();

//     expect(tree).toHaveStyleRule(
//       'font-family',
//       theme.fonts.sansSerif.replace(/\s+(?=([^"]*"[^"]*")*[^"]*$)/g, '')
//     ); // replace all whitespace not inside of double-quotes to properly match style rule output
//   });
// });
