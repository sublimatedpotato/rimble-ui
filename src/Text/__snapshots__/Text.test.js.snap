// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component sanity matches default snapshot 1`] = `
.c1 {
  color: #3F3D4B;
}

.c1 {
  box-sizing: border-box;
}

.c0 {
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  text-align: left;
  color: #3F3D4B;
}

<div
  class="c0 c1"
  color="copyColor"
  font-family="sansSerif"
  font-size="2"
  font-weight="2"
/>
`;

exports[`Text component sanity matches themed snapshot 1`] = `
.c2 {
  color: #3F3D4B;
}

.c2 {
  box-sizing: border-box;
}

.c1 {
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  text-align: left;
  color: #3F3D4B;
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <div
    class="c1 c2"
    color="copyColor"
    font-family="sansSerif"
    font-size="2"
    font-weight="2"
  />
</div>
`;
